lib/trino-parquet/src/main/java/io/trino/parquet/reader/ShortDecimalColumnReader.java

```diff
@@ -72,14 +68,7 @@ else if (field.getDescriptor().getPrimitiveType().getPrimitiveTypeName() == INT6
         }
         else {
             byte[] bytes = valuesReader.readBytes().getBytes();
-            if (typeLength <= Long.BYTES) {
-                value = getShortDecimalValue(bytes);
-            }
-            else {
-                int startOffset = bytes.length - Long.BYTES;
-                checkBytesFitInShortDecimal(bytes, startOffset, trinoType);
-                value = getShortDecimalValue(bytes, startOffset, Long.BYTES);
-            }
+            value = ShortDecimalValueDecoder.getDecoder(typeLength).decode(bytes, 0);
```

Comment:
Have we benchmarked this properly? The proposed approach will make this callsite megamorphic, which will likely affect performance.

Reply:
Right, the plan is to solve that problem through batched column readers. I've added a JMH benchmark with profile pollution to confirm that the new code is still better than the existing one.

Reply:
Can you post the results?

However a JMH benchmark by itself doesn't mean much. Have you looked at what the VM is doing in each of the scenarios? What inlining decisions is it making?

Reply:
Results are part of the last commit, I've now switched to the simpler approach suggested by you